lib/fastifySession.js

```diff
@@ -18,11 +18,16 @@ function session (fastify, options, next) {
   fastify.decorate('decryptSession', (sessionId, request, callback) => {
     decryptSession(sessionId, options, request, callback)
   })
-  fastify.decorateRequest('sessionStore', options.store)
-  fastify.decorateRequest('session', {})
+  fastify.decorateRequest('sessionStore', null)
+  fastify.decorateRequest('session', null)
   fastify.decorateRequest('destroySession', destroySession)
   fastify.addHook('preValidation', preValidation(options))
   fastify.addHook('onSend', onSend(options))
+  fastify.addHook('onRequest', (req, reply, next) => {
+    req.sessionStore = options.store
+    req.session = {}
```

Comment:
minor but I think this line is not necessary since it always gets assigned to either a new session or an existing one by the `preValidation` hook right? (I could have missed something, if so feel free to ignore this)